### DISCUSSION TITLE
Is there a way to specify the haskell-language-server executable used?

### DISCUSSION THREAD

**Discussion by youwen5** [mrcjkb/haskell-tools.nvim#368] [Q&A]
Hi everyone, I'm pretty new to Haskell (and Nix), so sorry if this is a silly question.

I've been trying to set up a Hakyll site using [this template](https://github.com/rpearce/hakyll-nix-template). I got everything built and working, but when I actually open the code for editing, haskell-tools.nvim crashes. Upon reviewing the logs, this is likely because it calls `haskell-language-server-wrapper`, which tells it to run `haskell-language-server` from the installation handled by Mason, which can't pick up the local Haskell dependencies managed by Nix.

I managed to get it working in the meantime with a dirty hack (symlinking the nix store's executable to the Mason install location), but I'm hoping there's a much better way to do this. Checking the outputs of `which haskell-language-server` and `which haskell-language-wrapper`, I see that they are both in the nix store. Furthermore, running `haskell-language-server-wrapper` directly seems to be successful with no errors. However, haskell-tools seems to ignore the haskell-language-server executables provided by nix and directly use the system ones which fail.

Am I doing something wrong? How can I make this work with Nix?

**Comment by mrcjkb** [comment]
Yes, see `:h haskell-tools.config` and `:h haskell-tools.lsp.ClientOpts` :smile:

**Reply by juarpr** [reply]
hello, :h HaskellLspClientOpts not working.
E149: Lo siento, no hay ayuda para "HaskellLspClientOpts"

**Reply by mrcjkb** [reply]
it's changed to `haskell-tools.lsp.ClientOpts`

**Reply by juarpr** [reply]
> Se ha cambiado a `haskell-tools.lsp.ClientOpts`

Thanks a lot!
